### DISCUSSION TITLE
Migrating to Elsa 3

### DISCUSSION THREAD

**Discussion by uyarbtrlp** [elsa-workflows/elsa-core#4879] [Q&A]
Hello,

We have been using Elsa 2 on razor page (MVC) and angular client applications. When I examined the documentation of Elsa 3, I could only see the blazor client integration for the Elsa designer and also the react component on the Github. I couldn't see any examples on MVC and angular clients. I have two questions:

1- Is it available on Angular as a component like we had in Elsa 2? If it is, is it fully implemented?
2- Is it applicable on MVC (razor page) like we had before?

**Comment by glime-ai** [comment]
Based on the information provided in the context, for Elsa 3, there are currently no official Angular components available. The development team has expressed an interest in adding Angular wrappers, but due to limited resources, this has not yet been completed. They have suggested that contributions from the community would be welcome in this area. The existing wrappers for Elsa 3 are primarily for React, as indicated in the Storybook project linked in one of the GitHub issues.

Regarding the second question, there is no explicit mention of MVC (razor page) integration for Elsa 3 in the provided context. However, Elsa is a .NET library that can be integrated into any .NET application, including MVC applications. Custom activities could theoretically be created to handle MVC-specific tasks such as screen navigation, as suggested in one of the Stack Overflow questions. For more detailed guidance on integrating Elsa with MVC applications, it would be best to refer to the official Elsa documentation or reach out to the Elsa community for support.

**Comment by sfmskywalker** [comment]
Hi, for Elsa 3, there is a new application called Elsa Studio that is built with Blazor and consist of reusable components that you can embed in your own MVC Razor page application like you can do with the Elsa 2 designer.

These components are currently provided in the following formats:

- Blazor components
- React wrappers for custom elements that wrap native Blazor WASM components

Angular wrappers have yet to be added. As mentioned by glime-ai, we are writing these wrappers using a Storybook project, and help is more than welcome.
